### PR TITLE
Fixes ENYO-3033

### PR DESCRIPTION
**Pull request by ryanjduffy**
In order to eliminate string references to kindNames and avoid circular
dependencies between the private kinds used by RelationalModel, some
refactoring was required including some post-kind setup in the module's
index.js

Enyo-DCO-1.1-Signed-off-by: Ryan Duffy (ryan.duffy@lge.com)

### CODE REVIEW

File: src/RelationalModel/toOne.js [hunk 1]
```diff
@@ -65,9 +69,6 @@ var toOne = module.exports = kind(
 		// ensure we have a valid model constructor
 		if (typeof model == 'string') model = kind.constructorForKind(model);
```
Comment (aarontam) [inline src/RelationalModel/toOne.js]: Can we remove this, as well?

File: src/RelationalModel/Relation.js [hunk 2]
```diff
@@ -107,7 +115,7 @@ var Relation = module.exports = kind(
 	/**
 	* @private
 	*/
-	setRelated: function (related) {
+	setRelated: function (related, opts) {
```
Comment (aarontam) [inline src/RelationalModel/Relation.js]: Curious if this having this unused parameter here was intentional, to match the overriding done by its subkinds.

Comment (ryanjduffy) [inline src/RelationalModel/Relation.js]: It was intentional. Adding a note to that effect.

File: src/RelationalModel/Relation.js [hunk 3]
```diff
@@ -100,14 +101,25 @@ var Relation = module.exports = kind(
...
+	*/
 	getRelated: function () {
 		return this.related;
 	},
 
 	/**
+	* Sets the `related` model
+	*
+	* @param {[type]} related - The related model
+	* @param {[type]} [opts] - Used by subkinds to configure the behavior of the method
```
Comment (webOS101) [inline src/RelationalModel/Relation.js]: I think related should probably be {module:enyo/Model~Model}  and opts should be {Object} or, if we have a typedef form options, that.

File: src/RelationalModel/toMany.js [hunk 4]
```diff
@@ -130,15 +129,22 @@ var toMany = module.exports = kind(
...
+	*/
+	isRelated: function (related) {
+		return this.related.indexOf(related) >= 0;
+	},
 	
 	/**
 	* @private
 	*/
-	setRelated: function (data) {
+	setRelated: function (data, opts) {
```
Comment (webOS101) [inline src/RelationalModel/toMany.js]: Might as well comment this one, too!

File: src/RelationalModel/Relation.js (hunk 3 above)
Comment (ryanjduffy) [inline src/RelationalModel/Relation.js]: oops ...

**Pull request merged**